daemon/list.go

```diff
@@ -133,7 +133,18 @@ func (daemon *Daemon) Containers(config *ContainersConfig) ([]*types.Container,
 			ID:    container.ID,
 			Names: names[container.ID],
 		}
-		newC.Image = container.Config.Image
+
+		img, err := daemon.Repositories().LookupImage(container.Config.Image)
+		if err != nil {
+			// If the image can no longer be found by its original reference,
+			// it makes sense to show the ID instead of a stale reference.
+			newC.Image = container.ImageID
```

Comment:
I'm not sure you need to call `LookupImage`. I set up three containers and printed out the values of `Container.ImageID` and `Container.Config.Image`. I think by comparing these you can determine the correct result:

This one was invoked with no tag and the image name still matches the correct image.
```
Container.image 8c2e06607696bd4afb3d03b687e361cc43cf8ec1a4a725bc96e39f05ba97dd55
Container.Config.image busybox
```

This one has had the tag moved:
```
Container.ImageID 2485b0f8995167b796a7ab4696e842c08bd27d58febb3ba0e935a4e2c1b2807d
Container.Config.Image 2485b0f8995167b796a7ab4696e842c08bd27d58febb3ba0e935a4e2c1b2807d
```

Here the image name still points to the right image and the container was invoked with a tag:
```
Container.ImageID ec64a324e9ccfbd431deed183c2a34cd3ded0704d763378d71aebbf08c483f73
Container.Config.Image darron/centurion-test:latest
```